File: app/src/main/java/com/waveneuro/ui/dashboard/DashboardCommand.java
package com.waveneuro.ui.dashboard;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;

import com.asif.abase.base.NavigationCommand;
import com.asif.abase.injection.qualifier.ActivityContext;

import javax.inject.Inject;

public class DashboardCommand extends NavigationCommand {

    private final Context mContext;

    @Inject
    public DashboardCommand(@ActivityContext Context mContext) {
        this.mContext = mContext;
    }

    @Override
    public void navigate() {
        Intent intent = new Intent(this.mContext, HomeActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        this.mContext.startActivity(intent);
        if (this.mContext instanceof Activity)
            ((Activity) this.mContext).finish();
    }
}

File: app/src/main/java/com/waveneuro/ui/dashboard/DashboardCommand.kt
package com.waveneuro.ui.dashboard

import android.app.Activity
import android.content.Context
import android.content.Intent
import com.asif.abase.base.NavigationCommand
import com.asif.abase.injection.qualifier.ActivityContext
import javax.inject.Inject

class DashboardCommand @Inject constructor(
    @ActivityContext private val context: Context
) : NavigationCommand() {

    override fun navigate() {
        val intent = Intent(context, DashboardActivity::class.java)
        intent.flags =
            Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
        context.startActivity(intent)
        if (context is Activity) context.finish()
    }

}